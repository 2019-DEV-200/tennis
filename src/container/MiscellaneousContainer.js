// Declaration of node module at the top
import React from 'react';
import { page_Title, player_A, player_B } from '../constant/scoreBoard.constant';
import ScoreBoardComponent from '../components/scoreBoardComponent';
import MatchUpdateComponent from '../components/matchUpdateComponent';
import { MainContainer, PlayerA, PlayerB } from './miscellaneous.Styled';

class MiscellaneousContainer extends React.Component {
	constructor(props) {
		super(props);
		// Declaration of state variables so as to hold
		this.state = {
			playerAScore: 0,
			playerBScore: 0,
			numberABtnClick: 1,
			numberBBtnClick: 1,
			gameOver: false
		};
	}

	// Declaration of life cycle event after render
	componentDidMount() {
		document.title = page_Title;
	}

	// When player A button clciked
	playerAHitsStroke(numberofAHits) {
		const { playerAScore, playerBScore, numberABtnClick, numberBBtnClick } = this.state;
		if (
			(numberABtnClick >= 4 && numberBBtnClick <= 3) ||
			(playerAScore === 40 && playerBScore === 40 && numberABtnClick - numberBBtnClick >= 1)
		) {
			this.setState({ gameOver: true });
		}
		if (numberofAHits === 1) {
			this.setState({ playerAScore: 15 });
		} else if (numberofAHits === 2) {
			this.setState({ playerAScore: 30 });
		} else if (numberofAHits === 3) {
			this.setState({ playerAScore: 40 });
		}
		// State value is incremented by 1
		this.setState({ numberABtnClick: this.state.numberABtnClick + 1 });
	}

	playerBHitsStroke(numberofBHits) {
		const { playerAScore, playerBScore, numberABtnClick, numberBBtnClick } = this.state;
		if (
			(numberBBtnClick >= 4 && numberABtnClick <= 3) ||
			(playerAScore === 40 && playerBScore === 40 && numberBBtnClick - numberABtnClick >= 1)
		) {
			this.setState({ gameOver: true });
		}
		if (numberofBHits === 1) {
			this.setState({ playerBScore: 15 });
		} else if (numberofBHits === 2) {
			this.setState({ playerBScore: 30 });
		} else if (numberofBHits === 3) {
			this.setState({ playerBScore: 40 });
		}
		// State value is incremented by 1
		this.setState({ numberBBtnClick: this.state.numberBBtnClick + 1 });
	}

	render() {
		// Destructuring of state
		const { playerAScore, playerBScore, numberABtnClick, numberBBtnClick, gameOver } = this.state;
		return (
			<React.Fragment>
				<MainContainer>
					{/* Passnig Props to scoreBoard Component */}
					<ScoreBoardComponent playerAScore={playerAScore} playerBScore={playerBScore} />

					<PlayerA
						label={player_A}
						className={`redBtn`}
						onClick={() => this.playerAHitsStroke(numberABtnClick)}
						disabled={gameOver}
					/>
					<PlayerB
						label={player_B}
						className={`greenBtn`}
						onClick={() => this.playerBHitsStroke(numberBBtnClick)}
						disabled={gameOver}
					/>
					{/* Passnig Props to Match Update Component */}
					<MatchUpdateComponent
						playerAScore={playerAScore}
						playerBScore={playerBScore}
						numberABtnClick={numberABtnClick}
						numberBBtnClick={numberBBtnClick}
					/>
				</MainContainer>
			</React.Fragment>
		);
	}
}

export default MiscellaneousContainer;
