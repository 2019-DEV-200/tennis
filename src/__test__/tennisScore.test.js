import React from 'react';
import renderer from 'react-test-renderer';
import {
	scorePoints,
	playerA_InitialScore,
	playerB_InitialScore,
	playerB_FirstPoint,
	playerA_ThirdPoint,
	playerB_ThirdPoint
} from '../constant/scoreBoard.constant';
import MiscellaneousContainer from '../container/MiscellaneousContainer';

// This test case will show valid score points
describe('Tennis score board points test case', () => {
	it('Should contain valid score points e.g 30', () => {
		expect(scorePoints).toContain(30);
	});
	it('Should not contain valid score points e.g 3', () => {
		expect(scorePoints).not.toContain(3);
	});
});

// Initial Score of both players same
describe('Initial score of both players test case', () => {
	it('Should have initial score to 0', () => {
		expect(playerA_InitialScore).toEqual(playerB_InitialScore);
	});
});

// Initial Score of both players is same
// if you change initial values then this test case will fail
describe('Initial Score is not same for both the players', () => {
	it('Initial score is not same for both the players', () => {
		expect(playerA_InitialScore).not.toEqual(playerB_FirstPoint);
	});
});

// if score of both players is three points then it will be deuce
describe('Both players score i.e 30', () => {
	it('Should show deuce when both players have 40 score points', () => {
		expect(playerA_ThirdPoint).toEqual(playerB_ThirdPoint);
	});
});

// Not Deuce condition
describe('Both players score three points test case', () => {
	it('Should show deuce when both players have 40 score points', () => {
		expect(playerA_ThirdPoint).not.toEqual(playerA_InitialScore);
	});
});

// snapshot test case for miscellaneous container
it('Should render miscellaneous container', () => {
	const tree = renderer.create(<MiscellaneousContainer />).toJSON();
	expect(tree).toMatchSnapshot();
});
