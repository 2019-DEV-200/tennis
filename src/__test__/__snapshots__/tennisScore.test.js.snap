// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render miscellaneous container 1`] = `
<div
  className="sc-dnqmqq efPCuS"
>
  <h1
    className="sc-bwzfXH eBfixS"
  >
    Match Score Board
  </h1>
  <table
    className="sc-bxivhb fWOJJx"
  >
    <tbody>
      <tr>
        <th
          className="sc-ifAKCX bliNJl"
        >
          Player A
        </th>
        <th
          className="sc-ifAKCX bliNJl"
        >
          Player B
        </th>
      </tr>
      <tr>
        <td
          className="sc-EHOje fxlDpa"
        >
          0
        </td>
        <td
          className="sc-EHOje fxlDpa"
        >
          0
        </td>
      </tr>
    </tbody>
  </table>
  <button
    className="sc-htoDjs isieRF sc-iwsKbI kVktMF redBtn"
    disabled={false}
    label="Player A"
    onClick={[Function]}
  >
    <span>
      Player A
    </span>
  </button>
  <button
    className="sc-htoDjs isieRF sc-gZMcBi fFtBOB greenBtn"
    disabled={false}
    label="Player B"
    onClick={[Function]}
  >
    <span>
      Player B
    </span>
  </button>
  <div
    className="sc-bZQynM bvcNwU"
  >
    <span
      className="sc-gzVnrw hjRexa"
    />
    Game Start
  </div>
</div>
`;
